src/cloudant/document.py

```diff
@@ -8,9 +8,23 @@
 import json
 import posixpath
 import urllib
+import requests
 
 from .errors import CloudantException
 
+# Actions for the update function below
+ACTIONS = {
+    "append_elem": lambda doc, field, value: doc[field].append(value),
+    "remove_elem": lambda doc, field, value: doc[field].remove(value),
+    "replace": lambda doc, field, value: doc.update({field: value})
+}
+class InvalidAction(Exception):
```

Comment:
So I think I would probably just make the action a callable so that people can customise it and use these as examples, maybe define them as module constants like APPEND_ELEM rather than forcing someone who wants to do something weird to add a function to a dictionary etc. 
Adding wrapper funcs like update_field_with_append for the common use cases would be convienient without limiting things.